src/main/java/org/truffleruby/core/module/ModuleFields.java

```diff
@@ -193,7 +195,12 @@ public void initCopy(RubyModule from) {
             this.constants.put(entry.getKey(), entry.getValue());
         }
 
-        this.classVariables.putAll(fromFields.classVariables);
+        for (Object key : fromFields.classVariables.getShape().getKeys()) {
+            DynamicObjectLibrary.getUncached().put(
+                    this.classVariables,
+                    key,
+                    DynamicObjectLibrary.getUncached().getOrDefault(fromFields.classVariables, key, null));
+        }
```

Comment:
@woess Isn't there a more direct way to do a (shallow) copy of a DynamicObject?
There used to be `DynamicObject#copy`, but I see nothing similar in DynamicObjectLibrary.